core/src/main/java/org/infinispan/xsite/ClusteredCacheBackupReceiver.java

```diff
@@ -200,13 +202,20 @@ private static PrivateMetadata internalMetadata(IracMetadata metadata) {
    }
 
    @Override
-   public final CompletionStage<Void> handleRemoteCommand(VisitableCommand command, boolean preserveOrder) {
+   public final CompletionStage<Object> handleRemoteCommand(ReplicableCommand command, boolean preserveOrder) {
+      //currently, it only handles sync xsite requests.
+      //async xsite requests are handle by the other methods.
+      assert !preserveOrder;
       try {
-         //currently, it only handles sync xsite requests.
-         //async xsite requests are handle by the other methods.
-         assert !preserveOrder;
-         //noinspection unchecked
-         return (CompletableFuture<Void>) command.acceptVisitor(null, defaultHandler);
+         if (command instanceof VisitableCommand) {
+            //noinspection unchecked
+            return (CompletionStage<Object>) ((VisitableCommand)command).acceptVisitor(null, defaultHandler);
+         } else if (command instanceof CacheRpcCommand) {
+            //noinspection unchecked
+            return (CompletionStage<Object>) ((CacheRpcCommand) command).invokeAsync(componentRegistry);
+         } else {
+            throw new UnsupportedOperationException("Only VisitableCommand and CacheRpcCommand is allowed!");
```

Comment:
Also, don't need to spend CPU throwing the exception. you can create the CF with it directly (`completedExceptionFuture()`)

Reply:
I personally don't think it is worth to add more code for a case that shouldn't ever happen.